### PR TITLE
Pr 111 x Displaced muon in BMTF and uGMT ( unpacker/emulator/packer)

### PR DESCRIPTION
**Pull request by rekovic**
#### PR description:

11_2_X:  L1T Displaced muons:  Pt from  unconstrainedVtx 
 - Unpacker/Emulator/Packer for KBMTF and uGMT (unpacker same for uGT)
 - Changes in DataFormat to accessors of the displaced-Pt and impact parameter

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

backport of #31608

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by silviodonato** [comment]
@rekovic @dinyar please update this PR including https://github.com/cms-sw/cmssw/pull/31848

**Comment by dinyar** [comment]
Hi @rekovic,

We need 
* 5c5d37583b792a12bfafa7a133a6ea2a6d83afc1
* 449b1038f2edecc4fb5e8dc6ae4d8cc14c5098e0
* acc2302071bf0d5f3c43763250bce639d69e2a92
as well as @silviodonato's fix from PR ##31848 to be in sync with #31608. 

Cheers,
Dinyar

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
@mrodozov @smuzaffar 
The above reported errors (like the one from 1000.0 step2 [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19454c/10309/runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log) have to do with accessibility of data and not with this PR.

**Comment by rekovic** [comment]
Excerpt from the log
```
[2020-10-27 10:10:52.189174 +0100][Error  ][XRootD            ] [eoscms.cern.ch:1094] Impossible to send message kXR_open (file: /eos/cms/store/user/cmsbuild/store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root, mode: 0660, flags: kXR_open_read kXR_async kXR_retstat ). Trying to recover.
%MSG-w XrdAdaptorInternal:  file_open 27-Oct-2020 10:10:52 CET pre-events
Failed to open file at URL root://eoscms.cern.ch:1094//eos/cms/store/user/cmsbuild/store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root.
%MSG
[2020-10-27 10:10:52.190619 +0100][Error  ][XRootD            ] [eoscms.cern.ch:1094] Unable to send the message kXR_open (file: /eos/cms/store/user/cmsbuild/store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root?tried=, mode: 0660, flags: kXR_open_read kXR_async kXR_retstat ): [FATAL] Connection error
27-Oct-2020 10:10:52 CET  Initiating request to open file root://cms-xrd-global.cern.ch//store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root
%MSG-w XrdAdaptorInternal:  file_open 27-Oct-2020 10:12:43 CET pre-events
Failed to open file at URL root://cms-xrd-global.cern.ch:1094//store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root?tried=+1098cms-xrd-global011098cms-xrd-global02.cern.ch.
%MSG
%MSG-w XrdAdaptorInternal:  file_open 27-Oct-2020 10:12:43 CET pre-events
Failed to open file at URL root://cms-xrd-global.cern.ch:1094//store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root?tried=.
%MSG
----- Begin Fatal Exception 27-Oct-2020 10:12:43 CET-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootInputFileSequence::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling XrdFile::open()
Exception Message:
Failed to open the file 'root://cms-xrd-global.cern.ch//store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root'
   Additional Info:
      [a] Calling RootInputFileSequence::initTheFile(): fail to open the file with name root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root
      [b] Input file root://cms-xrd-global.cern.ch//store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root could not be opened.
      [c] XrdCl::File::Open(name='root://cms-xrd-global.cern.ch//store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root', flags=0x10, permissions=0660) => error '[ERROR] Server responded with an error: [3011] No servers are available to read the file.
' (errno=3011, code=400). No additional data servers were found.
      [d] Last URL tried: root://cms-xrd-global.cern.ch:1094//store/data/Run2011A/MinimumBias/RAW/v1/000/165/121/0699429A-B37F-E011-A57A-0019B9F72D71.root?tried=
      [e] Problematic data server: cms-xrd-global.cern.ch:1094
      [f] Disabled source: cms-xrd-global.cern.ch:1094
----- End Fatal Exception -------------------------------------------------
```

**Comment by rekovic** [comment]
please test

(try again)

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
backport of #31608

**Comment by rekovic** [comment]
@jfernan2 
This updated PR is now fully in sync with the master PR.  We need it for the next MWGR.
Can you please review/sign.  Thanks.

File: EventFilter/L1TRawToDigi/plugins/implementations_stage2/BMTFPackerOutput.cc [hunk 1]
```diff
@@ -51,14 +51,14 @@ namespace l1t {
...
-	  //debug from here
-	  std::cout << "block id : " << block.header().getID() << std::endl;
-
-	  std::cout << "payload created : " << std::endl;
-	  for (auto &word : block.payload())  
-	    std::cout << std::bitset<32>(word).to_string() << std::endl;
-	  //debug up to here
-	*/
+      //debug from here
+      std::cout << "block id : " << block.header().getID() << std::endl;
```
Comment (jfernan2) [inline EventFilter/L1TRawToDigi/plugins/implementations_stage2/BMTFPackerOutput.cc:55]: despite the code is commented and it was already there, are couts allowed?

**Comment by jfernan2** [comment]
@rekovic since this PR is needed for next MWGR, we are testing it at P5 in DQM Playback

**Comment by jfernan2** [comment]
The test at P5 went fine, I am OK with the PR, I am not sure if the coding rules allow the couts even if commented

**Comment by dinyar** [comment]
Hi @jfernan2,

since this is a backport I suppose it would be best to keep it as is, but I could make a new PR to remove those lines. Would that be acceptable for you?

Cheers,
Dinyar

**Comment by jfernan2** [comment]
Sure

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**